contracts/REXOneWayMarket.sol

```diff
@@ -4,11 +4,14 @@ pragma solidity ^0.8.0;
 import "@uniswap/v2-periphery/contracts/interfaces/IUniswapV2Router02.sol";
 import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
+import "./libraries/REXUniswapV2Adapter.sol";
```

Comment:
Importing the library as of now. We can also deploy it separately if required. In that case, need to check if the caller context (delegatecall) is valid for ERC20 operations, otherwise will need to approve spending for the library contract.